packages/build-tools/tui/src/logs-standalone.ts

```diff
@@ -0,0 +1,111 @@
+import { fetchLogs, startServer, type TsLogExported } from "./logs-server.ts";
+
+// This is a standalone script that can be used to view logs from the collector.
+// Its purpose is to be used in a tmux session, and not as a part of the TUI.
+class LogsViewer {
+  private readonly pollInterval = 300; // ms
+  public isRunning = false;
+
+  formatLogEntry(entry: TsLogExported): string {
+    const timestamp = new Date(entry._meta.date).toLocaleTimeString();
+    const level = entry._meta.logLevelName;
+    // The log message is in the "0" field and contains ANSI color codes
+    const message = entry["0"];
+    const grey = (m: string) => `\x1b[90m${m}\x1b[0m`;
+
+    return `${grey(timestamp)} ${level} ${message}`;
+  }
+
+  async displayAndSaveLogs(logs: TsLogExported[]): Promise<void> {
+    // Only show new logs since last fetch\
+    const ddmmyyyy = new Date().toISOString().split("T")[0];
+    const fileBuffer: Record<string, string[]> = {};
+    for (const log of logs) {
+      const formatedLog = this.formatLogEntry(log);
+      const cleanLog = formatedLog.replace(this.ansiRegex(), "");
+      const namespace =
+        log["0"].replace(this.ansiRegex(), "").match(/^([\w-]+):\s/)?.[1] ??
+          "unknown";
+
+      // Write output to console.
+      console.log(formatedLog);
+      if (!fileBuffer[namespace]) {
+        fileBuffer[namespace] = [];
+      }
+      fileBuffer[namespace].push(cleanLog);
+    }
+
+    // Write output to file.
+    for (const [namespace, buffer] of Object.entries(fileBuffer)) {
+      await Deno.writeFile(
+        `./logs/${namespace}-${ddmmyyyy}.log`,
+        new TextEncoder().encode(buffer.join("\n") + "\n"),
+        { append: true },
+      );
+    }
```

Comment:
Do we want to write logs to a file instead of in-memory even in debug mode? Maybe the answer is yes, but I could also imagine the user accidentally using `console.log` in a loop and next they know they have a log file gigabytes in size

Reply:
I think this is probably more of an issue in debug builds that production, because in debug builds you're more likely to have a really spammy `console.log` to test something (which is when you accidentally write gigabytes of logs)

Reply:
https://github.com/PaimaStudios/paima-engine/commit/211ebcae46b05f2bbed84545a1ecb9045c101523
now using tsLogs to solve this